doc/design/python_interface.md

```diff
@@ -0,0 +1,101 @@
+# Python Interface Design
+
+## Variable as variable data type (not parameter)
+The class Variable is introduced to define the variables that are not parameters.
```

Comment:
The class Variable is introduced to define the variables, which are not only parameters but also the input data and intermediate outputs.